packages/sdk/src/modules/middlewareModule/utils/getRequestSender.ts

```diff
@@ -68,11 +68,20 @@ export const getRequestSender = (options: Options): RequestSender => {
       Accept: "application/json",
       ...defaultRequestConfig.headers,
     };
-    const mergedHeaders = {
-      ...defaultHeaders,
-      ...methodHeaders,
-      ...headers,
-    };
+    const mergedHeaders = Object.fromEntries(
+      Object.entries({
+        ...defaultHeaders,
+        ...methodHeaders,
+        ...headers,
+      }).reduce<[string, string | string[]][]>(
+        (headersSoFar, [key, value]) =>
+          // eslint-disable-next-line eqeqeq
+          key.toLocaleLowerCase() === "content-length" && value == "0"
```

Comment:
```suggestion
          // Removing Content-Lenght header if it equals 0 because it would crash Next.js server
          key.toLocaleLowerCase() === "content-length" && value == "0"
```